src/pymoca/tree.py

```diff
@@ -377,18 +388,38 @@ def build_instance_tree(orig_class: Union[ast.Class, ast.InstanceClass], modific
             # Fix component references to be one level deeper. E.g. applying a
             # modification "a.x = 3.0" on a symbol "a", will mean we pass
             # along a modification "x = 3.0" to the symbol's class instance.
+            # We should only do this if we are not (in)directly inheriting from a builtin.
+            inheriting_from_builtin = extends_builtin(c)
+            sym_mod = ast.ClassModification()
             for arg in sym_arguments:
                 if arg.value.component.indices:
                     raise Exception("Subscripting modifiers is not allowed.")
 
-                arg.value.component = arg.value.component.child[0]
+                if inheriting_from_builtin:
+                    for el_arg in arg.value.modifications:
+                        if not isinstance(el_arg, ast.ClassModification):
+                            # If the value is being set, we make a new class
+                            # modification with attribute name "value" that we
+                            # pick up later in modify_symbol()
+
+                            # TODO: Figure out if it's easier to directly do this
+                            # in the parser.
```

Comment:
Yeap that seems like the cleaner solution in the long run. Though I'd suggest we merge this for now, the TODO comment is in there after all.